fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -227,46 +224,68 @@ public String toString() {
      */
     public static EventType valueOf(final Integer i) {
         final EventType type = translation.get(i);
-        if (isNull(type)) {
+        if (type == null) {
             throw new IllegalArgumentException("Invalid event type: " + i);
         }
         return type;
     }
 
-
     /**
      * Convert a JCR Event to a FedoraEvent
+     *
      * @param event the JCR Event
      * @return a FedoraEvent
      */
     public static FedoraEvent from(final Event event) {
         requireNonNull(event);
         try {
-            @SuppressWarnings("unchecked")
             final Map<String, String> info = event.getInfo();
-
-            return new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
+            final FedoraEventImpl fedoraEvent = new FedoraEventImpl(valueOf(event.getType()), cleanPath(event),
                     event.getUserID(), event.getUserData(), event.getDate(), info);
-
+            getResourceTypes(event).forEach(fedoraEvent::addResourceType);
```

Comment:
See our conversation elsewhere. The event bus is not the right place to do that, if you insist on it being done. Externally-facing machinery like the JMS publishing gear is where that should be done, if it is to be done.